pymedphys/_gui/streamlit/mudensity-compare.py

```diff
@@ -814,33 +818,46 @@ def trf_input_method(patient_id="", key_namespace="", **_):
 
     headers
 
+    deliveries = cached_deliveries_loading(tables, delivery_from_trf)
+
+    individual_identifiers = [
+        f"{path.parent.parent.parent.parent.name} {path.parent.name}"
+        for path in selected_filepaths
+    ]
+
+    identifier = f"TRF ({individual_identifiers[0]})"
+
     """
     #### Corresponding Mosaiq SQL Details
     """
 
-    mosaiq_details = get_logfile_mosaiq_info(headers)
-    mosaiq_details = mosaiq_details.drop("beam_completed", axis=1)
-
-    mosaiq_details
-
-    patient_names = set()
-    for _, row in mosaiq_details.iterrows():
-        row
-        patient_name = utl_patient.convert_patient_name_from_split(
-            row["last_name"], row["first_name"]
+    try:
+        mosaiq_details = get_logfile_mosaiq_info(headers)
+        use_mosaiq = True
+    except KeyError:
+        use_mosaiq = False
+        st.write(
+            NoMosaiqAccess(
+                "Need Mosaiq access to determine patient name. "
+                "Patient name set to 'Unknown'."
+            )
         )
-        patient_names.add(patient_name)
+        patient_name = "Unknown"
 
-    patient_name = filter_patient_names(patient_names)
+    if use_mosaiq:
+        mosaiq_details = mosaiq_details.drop("beam_completed", axis=1)
```

Comment:
suggest extract method

Reply:
Yes, good call. Also, this whole file is too huge. ... it's unwieldly. It got that way due to streamlit not autoreloading on imports, so to keep the autoreloading I kept it in the same file. ...it's painful though.

I won't correct the file issue here, but I'll begin that by pulling this block out.

Reply:
So, I wanted to pull out `trf_input_method` into its own file, and then pull out this `use_mosaiq` block into its own function. I did that, but the flow on refactoring effect was large due to the need to pull out shared functions.

To make it easier to find, here is the function refactored out as suggested:

https://github.com/pymedphys/pymedphys/blob/853cca22de62e837286812408b9af7997d63f6f9/pymedphys/_gui/streamlit/mudensity/_trf.py#L34-L67